ui/app/src/components/SingleItemSelector/SingleItemSelector.tsx

```diff
@@ -127,7 +127,8 @@ const init: (props: SingleItemSelectorProps) => SingleItemSelectorState = (props
   limit: 10,
   total: 0,
   rootPath: props.rootPath,
-  currentPath: props.selectedItem?.path ?? props.rootPath
+  // If path is not a child of the rootPath (or the rootPath itself), then set the currentPath to be the rootPath
+  currentPath: (props.selectedItem?.path ?? '').includes(props.rootPath) ? props.selectedItem?.path : props.rootPath
```

Comment:
as a general rule, the SingleItemSelector will render the root if the item provided doesn't exist or if the item provided is not a child of the rootPath (or the rootPath itself)